packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1313,6 +1314,11 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
             _effectiveFocusNode.requestFocus();
           }
         };
+        handleDidLoseAccessibilityFocus = () {
+          if (_effectiveFocusNode.hasFocus) {
```

Comment:
This check is not necessary. `FocusNode` already checks for `hasFocus` and a couple of other things.